### DISCUSSION TITLE
Обновление версии Openvpn

### DISCUSSION THREAD

**Discussion by 3941701** [xtrime-ru/antizapret-vpn-docker#12] [Q&A]
Всех приветствую. Как-то был далек от всех этих VPN, линуксов и антизапретов, пока ютуб не начал тормозить) в итоге приходится все изучать в спешке. Но так и не понял, можно ли как-то обновить версию openvpn? Устанавливается версия 2.4.7 но начиная с версии 2.6.0 добавили крутую фишку Data Channel Offload (DCO) которая неплохо увеличивает скорость. Ведь сам по себе openvpn довольно требовательный к мощности железа и особенно актуально, когда клиент это дохлый роутер на старом проце

**Comment by xtrime-ru** [comment]
Я обновил openssl, но но написано, что Debian 10 не подеерживается: https://openvpn.net/as-docs/tutorials/tutorial--turn-on-openvpn-dco.html#set-vpn-tunnel-mtu--recommended-

Надо ждать когда базовый образ antizapret обновится. 

Либо не использовать базовый образ, а ставить с 0 на голую ubuntu/debian и настраивать. Но это будет трудоемкий процесс.

**Reply by xtrime-ru** [reply]
Переделал все на ubuntu24.04 и постаивил туда openvpn-dco-dkms
Версия openvpn свежая, с поддержкой DCO: 

```
OpenVPN 2.6.9 x86_64-pc-linux-gnu [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [PKCS11] [MH/PKTINFO] [AEAD] [DCO]
library versions: OpenSSL 3.0.13 30 Jan 2024, LZO 2.10
DCO version: N/A
Originally developed by James Yonan
Copyright (C) 2002-2023 OpenVPN Inc <sales@openvpn.net>
Compile time defines: enable_async_push=no enable_comp_stub=no enable_crypto_ofb_cfb=yes enable_dco=yes enable_dco_arg=yes enable_debug=yes enable_dependency_tracking=no enable_dlopen=unknown enable_dlopen_self=unknown enable_dlopen_self_static=unknown enable_fast_install=needless enable_fragment=yes enable_iproute2=no enable_libtool_lock=yes enable_lz4=yes enable_lzo=yes enable_maintainer_mode=no enable_management=yes enable_option_checking=no enable_pam_dlopen=no enable_pedantic=no enable_pkcs11=yes enable_plugin_auth_pam=yes enable_plugin_down_root=yes enable_plugins=yes enable_port_share=yes enable_selinux=no enable_shared=yes enable_shared_with_static_runtimes=no enable_silent_rules=no enable_small=no enable_static=yes enable_strict=no enable_strict_options=no enable_systemd=yes enable_unit_tests=no enable_werror=no enable_win32_dll=yes enable_wolfssl_options_h=yes enable_x509_alt_username=yes with_aix_soname=aix with_crypto_library=openssl with_gnu_ld=yes with_mem_check=no with_openssl_engine=auto with_sysroot=no
```

Протестил скорость на своем роутере Asus AX-55 - выдает 60 мбит во время просмотра 4K youtube видео. Проц при этом у него грузится на 25%. К сожалению со старой версией замеров не делал. 

Буду признателен за тесты новой версии контейнера!

**Reply by 3941701** [reply]
Установил, все работает стабильно, как и всегда, спасибо! Но теперь DCO не могу включить на клиентском устройстве) На сайте openvpn в туториале вот такая опция должна быть, но у меня ее нет ни на телефоне, ни в версии для ПК. Попробую на роутере после работы, но что-то подсказывает и там все будет не так просто, учитывая, что у меня еще падаван стоит
![uuid-c92e39e6-18aa-1366-8d46-b2f366c91ae8](https://github.com/user-attachments/assets/a076706d-2c0b-4e74-95e5-fa652f61d5a0)
